quic/s2n-quic-transport/src/transmission/application.rs

```diff
@@ -123,22 +125,27 @@ impl<'a, S: Stream, Config: endpoint::Config> Normal<'a, S, Config> {
             // soon as possible
             self.handshake_status.on_transmit(context);
 
-            //= https://www.rfc-editor.org/rfc/rfc9000#section-8.2
-            //# An endpoint MAY include other frames with the PATH_CHALLENGE and
-            //# PATH_RESPONSE frames used for path validation.
-            // prioritize PATH_CHALLENGE and PATH_RESPONSE frames higher than app data
-            self.path_manager.active_path_mut().on_transmit(context);
-
-            self.local_id_registry.on_transmit(context);
-
-            self.path_manager.on_transmit(context);
+            //= https://www.rfc-editor.org/rfc/rfc9221#section-5
+            //# DATAGRAM frames cannot be fragmented;
+            //
+            // We alternate between prioritizing filling the packet with datagrams
+            // and control frames. Datagrams cannot be fragmented across packets,
+            // so every other packet we give as much space as we can to send large datagrams.
+            if self.prioritize_datagrams {
```

Comment:
So I think these control frames are usually quite rare. The most common control frame are ACKs. It's probably a good ideal to include that in your logic

Reply:
Moving the ack_manager.on_transmit to this function technically changes behavior. Do we want it to come after sending handshake frames now? Also now it would be gated by this if statement: https://github.com/aws/s2n-quic/blob/main/quic/s2n-quic-transport/src/transmission/application.rs#L119

Reply:
We don't want to put it behind that if statement but we do need to gate ACK sending on datagram prioritization since those are going to be the control frames that most likely get in the way of sending full datagrams.